src/logion/controllers/secret_recovery.controller.ts

```diff
@@ -128,11 +142,111 @@ export class SecretRecoveryController extends ApiController {
             legalOfficerEMail: legalOfficer.userIdentity.email,
             userEmail: userIdentity?.email,
             data: {
-                secretName: secretRecoveryRequest.secretName,
                 legalOfficer,
                 walletUser: userIdentity,
-                walletUserPostalAddress: userPostalAddress
+                walletUserPostalAddress: userPostalAddress,
+                secret: secretRecoveryRequest,
             }
         }
     }
+
+    static fetchRecoveryInfo(spec: OpenAPIV3.Document) {
+        const operationObject = spec.paths["/api/secret-recovery/{id}/recovery-info"].put!;
+        operationObject.summary = "Fetch all info necessary for the legal officer to accept or reject secret recovery request.";
+        operationObject.description = "The authentication user must be a legal officers on current node";
+        operationObject.responses = getDefaultResponses("RecoveryInfoView");
+        setPathParameters(operationObject, { 'id': "The ID of the recovery request" });
+    }
+
+    @Async()
+    @HttpPut('/:id/recovery-info')
+    async fetchRecoveryInfo(_body: never, id: string): Promise<RecoveryInfoView> {
+        await this.authenticationService.authenticatedUserIsLegalOfficerOnNode(this.request);
+
+        const secretRecoveryRequest = await this.secretRecoveryRequestRepository.findById(id);
+        if(secretRecoveryRequest === null) {
```

Comment:
Should also test the auth user is the legal officer of the request.

Reply:
Indeed.